library/VTKExtensions/Rendering/vtkF3DRendererWithColoring.cxx

```diff
@@ -515,24 +508,44 @@ void vtkF3DRendererWithColoring::UpdateScalarBarVisibility()
 void vtkF3DRendererWithColoring::SetColoring(
   bool useCellData, const std::string& arrayName, int component, bool update)
 {
-  this->ArrayForColoring = nullptr;
-  this->DataForColoring = useCellData ? this->CellDataForColoring : this->PointDataForColoring;
-  this->ArrayIndexForColoring =
-    vtkF3DRendererWithColoring::FindArrayIndexForColoring(this->DataForColoring, arrayName);
   this->ComponentForColoring = component;
-  if (!this->DataForColoring ||
-    this->ArrayIndexForColoring >= this->DataForColoring->GetNumberOfArrays() ||
-    this->ArrayIndexForColoring < -1)
+  this->DataForColoring = useCellData ? this->CellDataForColoring : this->PointDataForColoring;
+
+  // TODO not really necessarry ?
+  if (!this->DataForColoring)
   {
-    F3DLog::Print(F3DLog::Severity::Error, "Invalid coloring values");
+    F3DLog::Print(F3DLog::Severity::Error, "Please set coloring attributes first");
     this->ArrayIndexForColoring = -1;
+    this->ArrayForColoring = nullptr;
+  }
+  else
+  {
+    this->ArrayIndexForColoring =
+      vtkF3DRendererWithColoring::FindArrayIndexForColoring(this->DataForColoring, arrayName);
+    if (this->ArrayIndexForColoring >= this->DataForColoring->GetNumberOfArrays() ||
+      this->ArrayIndexForColoring < -1)
+    {
+      F3DLog::Print(F3DLog::Severity::Error, "Invalid coloring values");
+      this->ArrayIndexForColoring = -1;
```

Comment:
Can you add a test to cover this case? It's just `--scalars=invalid` right?

Reply:
will be reworked in the cleanup PR which is incoming

Reply:
ok